tapfreighter/chain_porter.go

```diff
@@ -744,14 +744,13 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 				"proof file: %w", err)
 		}
 
-		// Hash proof locator.
-		hash, err := proofLocator.Hash()
```

Comment:
So before for proofs with the same asset ID + script key, we would end up clobbering them in this map, causing us to not write all relevant data to disk?

Reply:
Yes, though since these were just the passive assets within the same commitment, the script keys were forced to be unique, so no clobbering should have happened.

The actual "hack" this commit is cleaning up is that we use the hash of the `Locator` struct that isn't fully populated (only 2 of the 4 members). And then when accessing the map again further down in the call stack, we need to make sure we only fill the same 2 members to arrive at the same hash, otherwise we get a mismatch.
It was good for performance, but bad for understanding what's going on. When I looked for all instances of the `Locator` struct and set the `OutPoint` everywhere, this just broke.
IMO the new way of structuring the proofs has a slight performance hit but has way less chance of us screwing things up by accident.

Also, I'm planning on eliminating the "passive assets" concept soon anyway (in a future refactor PR spun out from the learnings of the TAP channel prototype), so this should be gone eventually.

Reply:
> Also, I'm planning on eliminating the "passive assets" concept soon anyway (in a future refactor PR spun out from the learnings of the TAP channel prototype), so this should be gone eventually.

Interesting, perhaps make a tracking issue to get more cross-pollination of the idea? Is this related to the `SIGHASH_NOINPUT` ideas, or other?